frontend/src/components/Contributions/ContributionForm.spec.js

```diff
@@ -48,64 +48,365 @@ describe('ContributionForm', () => {
     describe('empty form data', () => {
       describe('buttons', () => {
         it('has reset enabled', () => {
-          expect(wrapper.find('button[type="reset"]').attributes('disabled')).toBeFalsy()
+          expect(
+            wrapper.find('button[data-test="button-cancel"]').attributes('disabled'),
+          ).toBeFalsy()
         })
 
         it('has submit disabled', () => {
-          expect(wrapper.find('button[type="submit"]').attributes('disabled')).toBe('disabled')
+          expect(wrapper.find('button[data-test="button-submit"]').attributes('disabled')).toBe(
+            'disabled',
+          )
+        })
+      })
+    })
+
+    describe('dates', () => {
+      beforeEach(async () => {
+        await wrapper.setData({
+          form: {
+            id: null,
+            date: '',
+            memo: '',
+            amount: '',
+          },
+        })
+      })
+
+      describe('actual date', () => {
+        describe('same month', () => {
+          beforeEach(async () => {
+            const now = new Date().toISOString()
+            await wrapper.findComponent({ name: 'BFormDatepicker' }).vm.$emit('input', now)
+          })
+
+          describe('isThisMonth', () => {
+            it('has true', () => {
+              expect(wrapper.vm.isThisMonth).toBe(true)
+            })
+          })
+        })
+
+        describe('month before', () => {
+          beforeEach(async () => {
+            await wrapper
+              .findComponent({ name: 'BFormDatepicker' })
+              .vm.$emit('input', wrapper.vm.minimalDate)
+          })
+
+          describe('isThisMonth', () => {
+            it('has false', () => {
+              expect(wrapper.vm.isThisMonth).toBe(false)
+            })
+          })
+        })
+      })
+
+      // jest-environment-jsdom-sixteen
+      // testEnvironment: 'jest-environment-jsdom-sixteen',
+      // https://www.npmjs.com/package/jest-environment-jsdom-sixteen
+      // https://dev.to/philw_/problems-with-using-usefaketimersmodern-in-a-create-react-app-cra-project-with-jest-26-and-lodashs-debounce-function-3ohd
+      describe.skip('date in middle of year', () => {
+        describe('same month', () => {
+          beforeEach(async () => {
+            jest.useFakeTimers('modern')
```

Comment:
Here I get the error message:

`TypeError: Cannot read properties of undefined (reading 'useFakeTimers')`

Here is an hint why. But it didn't help after I followed the suggestion. Unfortunatelly …

Reply:
Did followup issue #2128